Src/Eiffel/eiffel/byte_code/visitor/assertion_breakable_slot_strategy.e

```diff
@@ -0,0 +1,481 @@
+﻿note
+	description: "Visitor for BYTE_NODE objects which count breakpoint slots in assertions."
+
+class
+	ASSERTION_BREAKABLE_SLOT_STRATEGY
+
+inherit
+	BYTE_NODE_ITERATOR
+		redefine
+			process_assign_b,
+			process_assert_b,
+			process_inv_assert_b,
+			process_require_b,
+			process_variant_b,
+			process_elsif_b,
+			process_elsif_expression_b,
+			process_if_b,
+			process_if_expression_b,
+			process_inspect_b,
+			process_instr_call_b,
+			process_loop_b,
+			process_loop_expr_b,
+			process_nested_b,
+			process_retry_b,
+			process_reverse_b,
+			process_check_b,
+			process_invariant_b,
+			process_guard_b
+		end
+
+	SHARED_BYTE_CONTEXT
+
+feature -- Status report
+
+	breakable_slot_count (a_node: BYTE_NODE; ctx: BYTE_CONTEXT): INTEGER
+			-- Does `a_node' contain an assignment to an attribute?
+		do
+			counter := 0
+			a_node.process (Current)
+			Result := counter
+
+			print (ctx.class_type.associated_class.name)
+			print (".")
+			print (ctx.current_feature.e_feature.name_8)
+			print (" -> " + a_node.generator + ": ")
+			print (Result)
+			print ("%N")
+
+		end
+
+feature {NONE} -- Implementation: access
+
+	counter: INTEGER
+
+	increment_counter (a_node: BYTE_NODE)
+		do
+			counter := counter + 1
+			if attached {ASSERT_B} a_node as a_assert and then attached a_assert.tag as t then
+				print (" - " + a_node.generator + " %"" + t + "%" [" + counter.out + "]%N")
+			else
+				print (" - " + a_node.generator + " [" + counter.out + "]%N")
+			end
+		end
+
+	generate_melted_debugger_hook_nested
```

Comment:
I'd remove the feature and calls to it.

Reply:
Why?

Reply:
Because it does not do anything. When reading the code, I asked myself - we are only counting breakpoint slots, why do we generate any hooks? Then I have to find this feature to see that it is empty.

Reply:
I think it does something ... for `a.b.c`  the bp index is the same, but this "...hook_nested" ... is to know if exception occurred on a.b or b.c ...

It is not yet really used , but the runtime is ready, and the debugger is displaying this info.

Reply:
I must be missing something. We are talking about finalized mode that does not work with the debugger at all.

Reply:
oups ... I forgot this was in this class.
so yes, I can remove it.